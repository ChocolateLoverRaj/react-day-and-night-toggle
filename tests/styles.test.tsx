import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {
    Switch,
    SwitchContainer,
    ContentWrapper,
    Circle,
    Star,
    Stars,
    ShootingStar,
    ShootingStarWrapper,
    Cloud,
    Clouds,
    Cloudpart
} from '../src/styles'


describe('Test of \'styled-components\' components:', () => {
    let testSize: number = 32
    let testShadow: boolean = true

    test('should <SwitchContainer> render:', () => {
        let component = renderer.create(<SwitchContainer size={testSize} />)
        expect(component.toJSON()).toMatchSnapshot()
        expect(component.root.props.size).toEqual(testSize)
    })
    test('should <Switch> render:', () => {
        let component = renderer.create(<Switch size={testSize} shadow={testShadow} />)
        expect(component.toJSON()).toMatchSnapshot()
        expect(component.root.props).toEqual({ size: testSize, shadow: testShadow })
    })
    test('should <ContentWrapper> render:', () => {
        let component = renderer.create(<ContentWrapper />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <Circle> render:', () => {
        let component = renderer.create(<Circle />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <Star> render:', () => {
        let component = renderer.create(<Star />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <Stars> render:', () => {
        let component = renderer.create(<Stars />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <ShootingStar> render:', () => {
        let component = renderer.create(<ShootingStar />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <ShootingStarWrapper> render:', () => {
        let component = renderer.create(<ShootingStarWrapper />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <Cloud> render:', () => {
        let component = renderer.create(<Cloud />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <Clouds> render:', () => {
        let component = renderer.create(<Clouds />)
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should <Cloudpart> render:', () => {
        let component = renderer.create(<Cloudpart />)
        expect(component.toJSON()).toMatchSnapshot()
    })

})
