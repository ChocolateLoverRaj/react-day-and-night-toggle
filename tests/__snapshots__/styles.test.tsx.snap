// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of 'styled-components' components: should <Circle> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <Cloud> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <Cloudpart> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <Clouds> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <ContentWrapper> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <ShootingStar> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <ShootingStarWrapper> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <Star> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <Stars> render: 1`] = `
<div
  className=""
/>
`;

exports[`Test of 'styled-components' components: should <Switch> render: 1`] = `
<div
  className=""
  size={32}
/>
`;

exports[`Test of 'styled-components' components: should <SwitchContainer> render: 1`] = `
<div
  className=""
  size={32}
/>
`;
